src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Index.cshtml

```diff
@@ -121,6 +124,23 @@
                                     </li>
                                 </ol>
                             </div>
+                            <div class="tab-pane" id="cpp-help">
+                                <ol>
+                                    <li>
+                                        Install clang 10 or later.
+                                    </li>
+                                    <li>
+                                        Run
+                                        <code>clang++ [input like sdk\storage\azure-storage-files-datalake\inc\azure\storage\files\datalake\datalake.hpp] -Xclang -ast-dump -I .\sdk\core\azure-core\inc -I .\sdk\storage\azure-storage-common\inc -I "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.27.29110\include" -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\ucrt" > az_datalake.ast</code>
```

Comment:
```
 clang++ sdk\storage\azure-storage-files-datalake\inc\azure\storage\files\datalake\datalake.hpp -Xclang -ast-dump -I .\sdk\core\azure-core\inc -I .\sdk\storage\azure-storage-common\inc -I "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.27.29110\include" -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\ucrt" > az_datalake.ast
sdk\storage\azure-storage-files-datalake\inc\azure\storage\files\datalake\datalake.hpp:6:10: fatal error: 'azure/storage/files/datalake/datalake_directory_client.hpp' file not found
#include "azure/storage/files/datalake/datalake_directory_client.hpp"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```
Didn't work for me.

Reply:
We might want to add a CMAKE target to get an AST file that would reference all the correct directories.

Reply:
The issue is where you run the tool from.  If you're not running from the DataLake folder then you'll need to include the DataLake inc folder.
`-I .\sdk\core\azure-storage-datalake\inc`

I'm supportive of a simplified CMake solution as a future improvement

Reply:
>The issue is where you run the tool from. If you're not running from the DataLake folder then you'll need to include the DataLake inc folder. -I .\sdk\core\azure-storage-datalake\inc

Still doesn't work:

```
clang++ sdk\storage\azure-storage-files-datalake\inc\azure\storage\files\datalake\datalake.hpp -Xclang -ast-dump -I .\sdk\core\azure-core\inc -I .\sdk\storage\azure-storage-common\inc -I "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.27.29110\include" -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\ucrt" -I .\sdk\core\azure-storage-datalake\inc > az_datalake.ast
sdk\storage\azure-storage-files-datalake\inc\azure\storage\files\datalake\datalake.hpp:6:10: fatal error: 'azure/storage/files/datalake/datalake_directory_client.hpp' file not found
#include "azure/storage/files/datalake/datalake_directory_client.hpp"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

Reply:
I think the datalake path you gave is incorrect for the master branch but even when using a correct one ` .\sdk\storage\azure-storage-files-datalake\inc\` it still doesn't work:

```
 clang++ sdk\storage\azure-storage-files-datalake\inc\azure\storage\files\datalake\datalake.hpp -Xclang -ast-dump -I .\sdk\core\azure-core\inc -I .\sdk\storage\azure-storage-common\inc -I "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.27.29110\include" -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\ucrt" -I .\sdk\storage\azure-storage-files-datalake\inc\ > az_datalake.ast
In file included from sdk\storage\azure-storage-files-datalake\inc\azure\storage\files\datalake\datalake.hpp:6:
.\sdk\storage\azure-storage-files-datalake\inc\azure/storage/files/datalake/datalake_directory_client.hpp:8:10: fatal error: 'azure/identity/client_secret_credential.hpp' file not found
#include "azure/identity/client_secret_credential.hpp"

```

Can you please share a command line that works without errors for you?

Reply:
Let me recheck and update.

Reply:
Ah yes we have added the indentity module in master.  @praveenkuttappan can you update to include identity 
`-I .\sdk\core\azure-identity\inc`

Reply:
Event then it doesn't work

```
 clang++ sdk\storage\azure-storage-files-datalake\inc\azure\storage\files\datalake\datalake.hpp -Xclang -ast-dump -I .\sdk\core\azure-core\inc -I .\sdk\storage\azure-storage-common\inc -I "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.27.29110\include" -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\ucrt" -I .\sdk\storage\azure-storage-files-datalake\inc\ -I .\sdk\identity\azure-identity\inc\ -v > az_datalake.ast

#include "azure/storage/blobs/blob_options.hpp"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

Reply:
DataLake isn't the easiest to build as it has dependencies on 3 other services.

Here is the command line for DataLake

`clang++ sdk\storage\azure-storage-files-datalake\inc\azure\storage\files\datalake\datalake.hpp -Xclang -ast-dump -I .\sdk\core\azure-core\inc -I .\sdk\storage\azure-storage-common\inc -I "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.27.29110\include" -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\ucrt" -I .\sdk\storage\azure-storage-files-datalake\inc\ -I .\sdk\identity\azure-identity\inc -I .\sdk\storage\azure-storage-blobs\inc > ast.ast`

Reply:
My laptop is getting reset due to access issues. I will update the exact command I tried once it is ready.